### PR TITLE
Added get/setResources method.

### PR DESCRIPTION
**Pull request by robbertstevens**

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=20047) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @robbertstevens. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by ihor-sviziev** [comment]
Hi @robbertstevens,
Why it's needed?

**Comment by robbertstevens** [comment]
@ihor-sviziev Autocompletion, also this method is necessary if you make Authorization rules by code. If it's not available in autocompletion it is annoying

File: app/code/Magento/Authorization/Model/Rules.php [hunk 1]
```diff
@@ -19,6 +19,7 @@
  * @method \Magento\Authorization\Model\Rules setAssertId(int $value)
  * @method string getPermission()
  * @method \Magento\Authorization\Model\Rules setPermission(string $value)
+ * @method setResources(array $resource)
```
Comment (orlangur) [inline app/code/Magento/Authorization/Model/Rules.php]: Please add return type here and add corresponding getter in form `int[]` or something - as specific as possible.

As always, please do amend commit and force push.

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Authorization/Model/Rules.php [hunk 2]
```diff
@@ -19,6 +19,8 @@
  * @method \Magento\Authorization\Model\Rules setAssertId(int $value)
  * @method string getPermission()
  * @method \Magento\Authorization\Model\Rules setPermission(string $value)
+ * @method string[] getResources()
+ * @method \Magento\Authorization\Model\Rules setResources(string[] $resource)
```
Comment (orlangur) [inline app/code/Magento/Authorization/Model/Rules.php:23]: Should be `resources`, I believe? Please squash changes into single commit and force push.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
Hi @robbertstevens , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**